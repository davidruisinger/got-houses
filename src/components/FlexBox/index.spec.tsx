import 'jest-styled-components'
import React from 'react'
import testRenderer from 'react-test-renderer'
import FlexBox from './index'

describe('<FlexBox />', () => {
  const tree = testRenderer.create(<FlexBox />).toJSON()
  it('matches the snapshot', () => {
    expect(tree).toMatchSnapshot()
  })
  it('has the correct default props', () => {
    expect(tree).toHaveStyleRule('align-items', 'stretch')
    expect(tree).toHaveStyleRule('align-content', 'stretch')
    expect(tree).toHaveStyleRule('flex-direction', 'row')
    expect(tree).toHaveStyleRule('justify-content', 'flex-start')
    expect(tree).toHaveStyleRule('flex-wrap', 'nowrap')
  })
})

describe('<FlexBox ...customProps />', () => {
  const tree = testRenderer
    .create(
      <FlexBox
        alignItems="center"
        alignContent="center"
        direction="column"
        justifyContent="center"
        wrap="wrap"
      />
    )
    .toJSON()
  it('matches the snapshot', () => {
    expect(tree).toMatchSnapshot()
  })
  it('has the correct custom props', () => {
    expect(tree).toHaveStyleRule('align-items', 'center')
    expect(tree).toHaveStyleRule('align-content', 'center')
    expect(tree).toHaveStyleRule('flex-direction', 'column')
    expect(tree).toHaveStyleRule('justify-content', 'center')
    expect(tree).toHaveStyleRule('flex-wrap', 'wrap')
  })
})
