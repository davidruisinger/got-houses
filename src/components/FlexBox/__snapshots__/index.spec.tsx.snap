// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlexBox ...customProps /> matches the snapshot 1`] = `
.c0 {
  width: 100%;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
}

<div
  className="c0"
  direction="column"
  wrap="wrap"
/>
`;

exports[`<FlexBox /> matches the snapshot 1`] = `
.c0 {
  width: 100%;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
}

<div
  className="c0"
  direction="row"
  wrap="nowrap"
/>
`;
